### ISSUE TITLE
Camera Photo Upload can no longer see camera folders.

### ISSUE THREAD

**Issue by wizkid057** [haiwen/seadroid#974]
Galaxy Fold Z3

After updating the app to 2.3.3, the camera auto upload stopped working.  Upon investigating, it appears the folders where new photos are stored isn't even listed as an option anymore in the list of folders to sync.

Some kind of new permission issue?  I don't recall the version I updated from, but it was probably > a year old but was working perfectly fine previously.

The "Backup Folder" feature can see the folder, however.

**Comment by gitouche-sur-osm** [comment]
Same here. Samsung Galaxy A6 (SM-A600FN). Android 10.

**Comment by zhwanng** [comment]
When using the camera auto-upload feature, selecting a folder backup is optional, if you do not select a folder, it will get the media data of the album by default.

You can turn off the "Custom Upload Albums" switch in the settings page.

**Comment by gitouche-sur-osm** [comment]
> When using the camera auto-upload feature, selecting a folder backup is optional, if you do not select a folder, it will get the media data of the album by default.
> 
> You can turn off the "Custom Upload Albums" switch in the settings page.

This is how I used before 2.3 and it worked perfectly.

While trying to find out why it stopped working, this is when I noticed (through the custom upload album) that seafile can't access the default camera album anymore. It's just not listed.

**Comment by wizkid057** [comment]
> When using the camera auto-upload feature, selecting a folder backup is optional, if you do not select a folder, it will get the media data of the album by default.
> 
> You can turn off the "Custom Upload Albums" switch in the settings page.

Yep, understand how it's _supposed_ to work, but the Camera album doesn't show up for camera uploads anymore as noted by @gitouche-sur-osm above as well.

**Comment by iBigQ** [comment]
Sounds that I had the same issue. Try to go to Android Settings->Apps->Seafile and check the app permissions. Everything seemed fine to me but I revoked media/disk permissions anyway and then enabled them afterwards. After force-stopping the app, it was able to "see" the foto album

**Comment by VanLampe** [comment]
> Sounds that I had the same issue. Try to go to Android Settings->Apps->Seafile and check the app permissions. Everything seemed fine to me but I revoked media/disk permissions anyway and then enabled them afterwards. After force-stopping the app, it was able to "see" the foto album

I had the same problem. Your solution worked for me as well.

**Comment by zhwanng** [comment]
Yes, I found it. When I turn off STORAGE permission for Seafile APP in Android Settings, I can't see Photo album either. When I turn on permissions it is visible. I guess it must be a permission issue.

However, when the app opens the home page, it shows a STORAGE permission request dialog, did you click the "Allow" button? Or do you have no dialog at all?

**Comment by iBigQ** [comment]
I have been running the seafile App for a long time already. Also, I Made No Changes recently. The App did Not ask for permission recently and the permission setting in Android settings seemed fine. But still, the App behaved Like it did Not have the permission. So I agree, likely a permission issue. I have another Android Phone (older Android Version, also Pixel phone) that suddenly stopped Foto Upload with the Same issue. Hope you find the root cause.Am 10.06.2023 11:20 schrieb zhwanng ***@***.***>:
Yes, I found it. When I turn off STORAGE permission for Seafile APP in Android Settings, I can't see Photo album either. When I turn on permissions it is visible. I guess it must be a permission issue.
However, when the app opens the home page, it shows a STORAGE permission request dialog, did you click the "Allow" button? Or do you have no dialog at all?

**Comment by bertvandepoel** [comment]
I had been wondering why my camera upload stopped working about 1 month ago (last pics uploaded from 16 May). Tried restarting the server, restarting my phone, updating the server, checked if I was running the latest Seadroid, nothing helps. Then I wanted to re-setup Camera uploads and noticed that my Camera album and several others are missing. It seems I am far from the only one!

The fix proposed by @iBigQ worked for me too, though things had a different name for me: settings -> Apps -> App Management -> Seafile. Then the permission is called Photos and Videos (when you disable it, it will alsot take along Audio and Music or something like that). After re-enabling it, you will see a bunch of albums again, though to be honest I don't think those are actually all my albums. My guess is that Seadroid actually does have a bug that is being triggered. Specifically: it's fetching only a certain number of albums instead of all of them. I do hope that @zhwanng @maxiaoping or another member of Haiwen/Seafile will look into this!

**Comment by Josue-T** [comment]
Hello,

I confirm that the issue is a permission issue. On my side on 2 phone I detected the same issue. Since this spring camera directory was not any more shown for camera photo upload. And doing again the camera upload setup process didn't fix the issue. The only solution which worked for me was to enable the backup feature and here I had a popup which asked for full directory access. And after that I was able to see the camera directory from camera photo upload.

So maybe the fix could be to request the same permission than for the backup feature.

**Comment by wizkid057** [comment]
I've been using folder backup as an alternative to the broken Camera Uploads since this bug presented, and while this kind of works it has its own issues (https://github.com/haiwen/seadroid/issues/973) and despite this working it does NOT make anything accessible via the Camera Uploads feature.

**Comment by zhwanng** [comment]
I'm tracking this issue, Can you provide some screen recordings or screenshots or other logs? So that help I can fix this.

**Comment by zhwanng** [comment]
Is there nothing here? Or is there just no "Camera" folder?
BTW, the system version is ?

![Screenshot_20230719_102145](https://github.com/haiwen/seadroid/assets/48609908/a361e07b-9a7e-416d-a585-cc51b8fd69e4)

**Comment by bertvandepoel** [comment]
For me, it showed some albums, but not all of them. It also showed certain albums with the same name (each only containing a few images, it seemed). Important albums like Camera were missing. It was as if Seadroid wasn't getting or requesting every album but only a limited number.  I also mentioned this in my comment earlier in this issue. 

I'm running OxygenOS 13.1, which is Android 13. 

Is there any more information you'd like me to supply?

**Comment by woj-tek** [comment]
Same problem here - only two (seemingly random) folders shown. And upload is not working (seafile 2.3.2 from Fdroid, Android 13 / OneUI 5.1)

**Comment by zhwanng** [comment]
@woj-tek Please update to the latest version(v2.3.5) and test this issue

**Comment by woj-tek** [comment]
@zhwanng it hasn't been released yet (https://github.com/haiwen/seadroid/releases/tag/2.3.5)? And given that I use f-droid I would have to wait for it to be updated (after the release is made).

**Comment by zhwanng** [comment]
released -> [https://github.com/haiwen/seadroid/releases/tag/2.3.5](https://github.com/haiwen/seadroid/releases/tag/2.3.5)

**Comment by zhwanng** [comment]
@woj-tek Did something go wrong?

**Comment by woj-tek** [comment]
@zhwanng Finally I was update to the 2.3.5 after #972 changes. The good thing is that I now see all folders and I'm able to select desired one but - nothing is uploaded. And "Upload info" section says only: "Done".
I tried creating new upload folder on remote, selecting it in "photo upload section" but without any change - still nothing got uploaded.

**Comment by zhwanng** [comment]
@woj-tek 

For the folder/album backup feature, we have some major updates coming in the future and you can wait for it.
A new version has just been updated -> [v2.3.6](https://github.com/haiwen/seadroid/releases/tag/2.3.6). You can download and use it.
We'll close this issue for now, and if it comes up again after a major update in the future, we'll turn it back on.

**Comment by bertvandepoel** [comment]
@zhwanng will this update be pushed to the Google Play store and f-droid? Both of them are still on 2.3.4 and are now two releases behind.

**Comment by zhwanng** [comment]
> @zhwanng will this update be pushed to the Google Play store and f-droid? Both of them are still on 2.3.4 and are now two releases behind.

google play is reviewing the new version, please wait for the review.

**Comment by wizkid057** [comment]
Will note that 2.3.6 still has the original issue and this is _not_ resolved.

Camera uploads still doesn't have the option to choose my actual camera folder. I can find it with "Folder Backup", but not "Camera Upload".

Additionally, for whatever reason Folder Backup doesn't work in the background anymore, despite having all power-save stuff disabled for the app and everywhere in my phone.  I've even tried locking Seafile as open and all also, with no success.  Nothing is backed up unless I explicitly open Seafile, and when I look at settings it will show the transfers happening all at once.

Becoming somewhat of a hassle at this point, so likely going to have to find some other solution.

**Comment by zhwanng** [comment]
@wizkid057 Please provide your mobile phone model and system version.

**Comment by zhwanng** [comment]
@woj-tek f-droid builds seadroid automatically and you can subscribe to its updates

**Comment by wizkid057** [comment]
> @wizkid057 Please provide your mobile phone model and system version.

Same as before. Samsung Galaxy Z Fold 3 SM-F926U.
Android 14

**Comment by zhwanng** [comment]
![WX20231219-104637](https://github.com/haiwen/seadroid/assets/48609908/802d68a0-83a5-44c4-b96f-66095958c5d8)
is here nothing to show?

**Comment by wizkid057** [comment]
> ![WX20231219-104637](https://private-user-images.githubusercontent.com/48609908/291460335-802d68a0-83a5-44c4-b96f-66095958c5d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NjE3MzgsIm5iZiI6MTcwMjk2MTQzOCwicGF0aCI6Ii80ODYwOTkwOC8yOTE0NjAzMzUtODAyZDY4YTAtODNhNS00NGM0LWI5NmYtNjYwOTU5NThjNWQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDA0NTAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmNTRlOTg2YjlmYWI5MzA3YmY3MjFhYjhmYWRiYmMzMDJlN2Q5ZTNjNThiYjY4YTIwNDgwZjc0NWEyMmIyNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ty2dzNe4UDm8iuODLHtfS0NOzjV8aY7Ck0CAdGDSNG4) is here nothing to show?

Folders with photos from all over my device show there (Downloads, images from misc apps, etc), but _not_ the actual Camera folder that has my photos.  It's quite odd.  I'm willing to chalk this up to something odd on my end, but that doesn't get me anywhere useful either.

**Comment by bertvandepoel** [comment]
I installed 2.3.6 manually since the Google Play store release is still not available. It did cause me to see some missing albums again. Sadly, it is also causing Seadroid to resync every image and video I had deleted. All my stuff I deleted from prior to halfway 2022 is now being reuploaded, mixed into missing albums. This means I had to increase my VPS disk size, which is not ideal.

Is this intended behaviour, a one time side-effect, or a regression bug?
